tests/robottelo/decorators/test_host.py

```diff
@@ -12,139 +14,197 @@
             u'Red Hat Enterprise Linux Server release 6 (Maipo)',
             u'RHEL6'
         )
 
-    def test_rhel_7_2(self):
+    def test_rhel_minor_version_parsing(self):
+        """Version semantic example: 1.2.3
+        1 is major
+        2 is minor
+        3 is patch
+        """
         self.assert_rhel_version(
             u'Red Hat Enterprise Linux Server release 7.2 (Maipo)',
             u'RHEL7.2'
         )
 
-    def test_rhel_7_2_1(self):
+    def test_rhel_patch_version_parsing(self):
+        """Version semantic example: 1.2.3
+        1 is major
+        2 is minor
+        3 is patch
+        """
         self.assert_rhel_version(
             u'Red Hat Enterprise Linux Server release 7.2.1 (Maipo)',
             u'RHEL7.2.1'
         )
 
     def test_cache(self):
+        """Test _get_host_os_version() is called only once, caching the host
+        version for next calls
+        """
         self._command.return_value.stdout = [
             u'Red Hat Enterprise Linux Server release 7.2.1 (Maipo)'
         ]
-        self.assertEqual(u'RHEL7.2.1', host._cached_host_os_version())
+        self.assertEqual(u'RHEL7.2.1', host._get_host_os_version())
         self._command.assert_called_once_with('cat /etc/redhat-release')
         self._command.return_value.stdout = [
             u'Doesnt matter because because its cached'
         ]
-        self.assertEqual(u'RHEL7.2.1', host._cached_host_os_version())
+        self.assertEqual(u'RHEL7.2.1', host._get_host_os_version())
         # if called more than once cache didn't worked
         self._command.assert_called_once_with('cat /etc/redhat-release')
 
     @mock.patch('robottelo.decorators.host.LOGGER')
     def test_command_error(self, logger):
+        """Test error on ssh results logging a warning and version not
+        available
+        """
         cmd = SSHCommandResult(
             stdout=[],
             stderr=u'bash: generate: command not found\n',
             return_code=127, output_format=None
         )
         self._command.return_value = cmd
 
-        host._get_host_os_version()
+        os_version = host._get_host_os_version.__wrapped__()
+        self.assertEquals('Not Available', os_version)
         self._command.assert_called_once_with('cat /etc/redhat-release')
         logger.warning.assert_called_once_with(
             u'Host version not available: %r' % cmd)
 
     @mock.patch('robottelo.decorators.host.LOGGER')
     def test_command_parsing_error(self, logger):
+        """Test error on ssh results logging a warning if version not
+        available. It can parse other OS in future if needed.
+        """
         cmd = SSHCommandResult(
             stdout=[u'Fedora release 23 (Twenty Three)'],
             return_code=0
         )
         self._command.return_value = cmd
 
-        host._get_host_os_version()
+        os_version = host._get_host_os_version.__wrapped__()
+        self.assertEquals('Not Available', os_version)
```

Comment:
Prefer assertEqual without the `s`, `assertEquals` is a deprecated alias [1]

https://docs.python.org/2/library/unittest.html#deprecated-aliases